app/models/course/forum.rb

```diff
@@ -4,8 +4,46 @@ class Course::Forum < ActiveRecord::Base
 
   belongs_to :course, inverse_of: :forums
   has_many :topics, dependent: :destroy, inverse_of: :forum
+  has_many :posts, class_name: Course::Discussion::Post, through: :topics
   has_many :subscriptions, dependent: :destroy, inverse_of: :forum
 
+  # @!method self.with_topic_count
+  #   Augments all returned records with the number of topics in that forum.
+  scope :with_topic_count, (lambda do
+    joins { topics.outer }.
+      select { 'course_forums.*' }.
```

Comment:
hmm... Seems if I move this line, this sql will only select the count number without forums information.